### PR TITLE
Rule attribute to override timeout set by dbus

### PR DESCRIPTION
**Pull request by rpbranco**
This adds a new (modifying) rule attribute `override_dbus_timeout`, as requested in #1035.

The timeout priority is now:
1. Rule `override_dbus_timeout`
2. Dbus timeout
3. Rule `timeout`
  (4. Internal defaults)

### CODE REVIEW

File: src/dbus.c [hunk 1]
```diff
@@ -604,6 +604,9 @@ static struct notification *dbus_message_to_notification(const gchar *sender, GV
                 g_variant_unref(dict_value);
         }
 
+        // Set the dbus timeout
+        n->dbus_timeout = timeout < 0 ? timeout : ((gint64)timeout) * 1000;
```
Comment (fwsmit) [inline src/dbus.c:608]: Instead of storing a separate dbus_timeout variable, I'd just override the timeout.

File: src/rules.h [hunk 2]
```diff
@@ -29,6 +29,7 @@ struct rule {
 
         /* modifying */
         gint64 timeout; // this has to be the first modifying rule
+        gint64 override_dbus_timeout;
```
Comment (fwsmit) [inline src/rules.h:32]: You can then make this an `override_timeout` variable (set by the `override_dbus_timeout` setting). That can be used in `notification_init` after the rule processing to override the `timeout`.

**Comment by fwsmit** [comment]
Thanks for improving your PR. I think this is the right approach. The only comment I have is that the logic can be made a bit simpler. See the inline comments

**Comment by rpbranco** [comment]
Thank you for the fast reply! Regarding your suggestions, and before implementing them, how would you feel if I were to implement a rule attribute to allow filtering based on the dbus timeout?

I ask this before making any changes because, in order to implement this, it would be necessary to know the dbus timeout when applying the rules.

**Comment by fwsmit** [comment]
> Thank you for the fast reply! Regarding your suggestions, and before implementing them, how would you feel if I were to implement a rule attribute to allow filtering based on the dbus timeout?
> 
> I ask this before making any changes because, in order to implement this, it would be necessary to know the dbus timeout when applying the rules.

If that's the case, then yes don't remove it. Matching on dbus timeout would be great, although maybe not very useful on it's own. What's your use case? Maybe it's more useful in combination with #1040 to allow for `match_dbus_timeout =! -1` or something.

File: src/dbus.c (hunk 1 above)
Comment (fwsmit) [reply src/dbus.c:608]:
```suggestion
if (timeout >= 0)
        n->dbus_timeout =((gint64)timeout) * 1000;
```

If you want to use the dbus_timeout in the future, I'd still suggestion doing this.
Also, I think it's better to modify n->timeout in notification.c then.

**Comment by rpbranco** [comment]
I made the suggested changes.

Regarding my use case for the `match_dbus_timeout`, I have an application that sets the dbus timeout of some notifications to `0`. I would like to be able to match only these notifications.

Sorry for the delayed answer!

**Comment by fwsmit** [comment]
> I made the suggested changes.
> 
> Regarding my use case for the `match_dbus_timeout`, I have an application that sets the dbus timeout of some notifications to `0`. I would like to be able to match only these notifications.
> 
> Sorry for the delayed answer!

Great, thank you. The changes look good, so I'll merge it

**Pull request merged**